keyboards/converter/usb_usb/keymaps/bepo/keymap.c

```diff
@@ -0,0 +1,216 @@
+/*
+	Transform a BÉPO keyboard into a standart AZERTY keyboard based on a iso layoutPO
+	
+	Use only when you cannot install the native BÉPO driver
+	as for instance if you are not allowed to install program / drivers on a business computer
+	Your windows has to be set with a French AZERTY input method (laptop keyboard will stay in azerty for instance)
+	
+	NOTES:
+	* All special code are specific to Windows (ALT+wincode input method)
+	* Many dead letters are not implemented (lack of alt-code for that) - only the one available on an azerty (~, ^, ¨, ` )
+	* fine insecable input a "standard" insecable space, to adapt to Word corectly. Easy to adapt (ALT+8239 instead of AL+255)
+	* Compile with the the option "LTO_ENABLE = yes" in your rules.mk, to reduce the size of the hex
+	
+	At any time, swith possible between bépo and AZERTY:
+	* Shift + AltGr + "^ (azerty)/z (bépo)" key = AZERTY layout
+	* Shift + AltGr + "$ (azerty)/w (bépo)" key = BÉPO emulation layout
+	Last mode is not persistent on the Hasu converter (lack of persistent memory?). Default keyboard is bépo at power-on
+	
+*/
+#include QMK_KEYBOARD_H
+#include <keymap_french.h>
+
+#define WINALT1(A) {SEND_STRING(SS_LALT(SS_TAP(X_KP_ ## A))); return false;}

```

Comment:
All of this could probably be replaced by the Unicode feature - I believe it supports the Windows Alt codes.

Reply:
Probably... but was not working on my business laptop. Seems that registry value HKEY_Current_User/Control Panel/Input Method/EnableHexNumpad might be updated but I'm not allowed to change it. Behaviour with ALT + 4 digits codes (in decimal) is not perfect - as for instance works fine in Word but only for some on Excel - but hopefully with the file above all French letters are working perfectly, as well as the full punctuation (including many specific French rules). In my case, I type in French and in English (with many mistakes ;-)), and the set of codded letters in this "emulation" is enough at work. At home, I can use the native drivers on any OS, I'm admin on those ones. I will still investigate the hints provided here to try to implement as much as possible, but I still have to stick to the initial constraint: I cannot change whatever to the OS settings).

Reply:
That's fair. One other potential improvement would be to turn these macros into a function that calls [`hex_to_keycode()`](https://github.com/qmk/qmk_firmware/blob/master/quantum/quantum.c#L727-L736) on the arguments, rather than using the preprocessor concatenation. You may also have to reimplement that function though, to make it use the keypad numbers rather than the number row.

Reply:
I had a look on that function (and some alternatives), but indeed I move to a "pre-processor" way to minimize the processing time, and also because the ALT+code in Windows is tricky (several pages, some with a leading 0, some without, better support for "short-codes" instead of the UTF-8 ones when existing...). So, at the end I preferred to test each entry manually and put them in this way in the code for a direct execution. Of course, it take more place in the hex, so it is only valid because I didn't reach the limit. Note: as I will work on the support of the trackpad when I will have time, I might have to find some ways to reduce the hex size...

Reply:
Seems I made a mistake when posting my answer (not yet fluent in GitHub commands... implementing a hex_to_keycode function was my first attempt, and some have develop such functions for Alt Win codes.. but they are long and complex (because of Windows code pages...). At the end, as the code fit in the converter with this "pre-compiled code" I prefer this option to minimize the execution process, as no calculation have to be performed at execution time.
Later I intend to work on the support of the integrated trackpad of my Lenovo keyboard, I might need more space in the hex and have to follow this functional approach.